ggml/src/ggml-cpu/amx/amx.h

```diff
@@ -1,20 +1,9 @@
 #include "ggml-backend.h"
 #include "ggml-cpu-impl.h"
 
-#ifdef __cplusplus
-extern "C" {
-#endif
+// GGML internal header
 
+// if defined(GGML_USE_CPU_AMX) ?
```

Comment:
Oh yes. Actually it was to ask a question.
I wondered if at this point it would not be better to be able to disable this "register" at compile time without it being managed by the CPU features. 
This would allow to manage the AVX512 which is coded on CPUs without AMX, and to simply disable it on those that support it.
But it is rather for another PR.

But it is more to be unified with `GGML_USE_CPU_AARCH64` / `GGML_USE_CPU_AMX`: 
```c++
#if defined(__AMX_INT8__) && defined(__AVX512VNNI__)
        if (ggml_backend_amx_buffer_type()) {
            bufts.push_back(ggml_backend_amx_buffer_type());
        }
#endif

#ifdef GGML_USE_CPU_AARCH64
        if (ggml_backend_cpu_aarch64_buffer_type()) {
            bufts.push_back(ggml_backend_cpu_aarch64_buffer_type());
        }
#endif
```